bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -308,13 +311,21 @@ public Iterable<HistoricItem> query(FilterCriteria filter) {
                         start = end;
                     }
                 } else {
-                    throw new UnsupportedOperationException("rrd4j does not allow querys without a begin date, "
-                            + "unless order is descending and a single value is requested");
+                    throw new UnsupportedOperationException(
+                            "rrd4j does not allow querys without a begin date, unless order is descending and a single value is requested");
                 }
             } else {
                 start = filter.getBeginDate().toInstant().getEpochSecond();
             }
 
+            // do not call method {@link RrdDb#createFetchRequest(ConsolFun, long, long, long)} if start > end to avoid
+            // an IAE to be thrown
+            if (start > end) {
+                logger.warn("Could not query rrd4j database for item '{}': start ({}) > end ({})", itemName, start,
+                        end);
+                return List.of();
+            }
```

Comment:
We could consider throwing `IllegalArgumentException` by ourselves here, since this would now be a bug on caller side, where previously it was an internal bug within this method. But let's leave something for future improvement also. 🙂